src/Elliptic/Systems/Xcts/Events/ObserveAdmIntegrals.cpp

```diff
@@ -0,0 +1,102 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/Systems/Xcts/Events/ObserveAdmIntegrals.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/AreaElement.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/ProjectToBoundary.hpp"
+#include "NumericalAlgorithms/LinearOperators/DefiniteIntegral.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "PointwiseFunctions/Xcts/AdmLinearMomentum.hpp"
+
+namespace Events {
+
+void compute_linear_momentum(
+    gsl::not_null<tnsr::I<double, 3>*> result,
+    const Scalar<DataVector>& conformal_factor,
+    const tnsr::ii<DataVector, 3>& spatial_metric,
+    const tnsr::II<DataVector, 3>& inv_spatial_metric,
+    const tnsr::ii<DataVector, 3>& extrinsic_curvature,
+    const Scalar<DataVector>& trace_extrinsic_curvature,
+    const InverseJacobian<DataVector, 3, Frame::ElementLogical,
+                          Frame::Inertial>& inv_jacobian,
+    const Mesh<3>& mesh, const Element<3>& element,
+    const DirectionMap<3, tnsr::i<DataVector, 3>>& faces_map_normal) {
+  // Initialize integral to 0
+  for (int I = 0; I < 3; I++) {
+    result->get(I) = 0;
+  }
+
+  // Skip elements not at the outer boundary
+  const size_t radial_dimension = 2;
+  const auto radial_segment_id = element.id().segment_id(radial_dimension);
+  if (radial_segment_id.index() !=
+      two_to_the(radial_segment_id.refinement_level()) - 1) {
+    return;
+  }
+
+  // Loop over external boundaries
+  for (auto boundary_direction : element.external_boundaries()) {
+    // Skip interfaces not at the outer boundary
+    if (boundary_direction != Direction<3>::upper_zeta()) {
+      continue;
+    }
+
+    // Project fields to the boundary
+    const auto& face_conformal_factor = dg::project_tensor_to_boundary(
+        conformal_factor, mesh, boundary_direction);
+    const auto& face_spatial_metric = dg::project_tensor_to_boundary(
+        spatial_metric, mesh, boundary_direction);
+    const auto& face_inv_spatial_metric = dg::project_tensor_to_boundary(
+        inv_spatial_metric, mesh, boundary_direction);
+    const auto& face_extrinsic_curvature = dg::project_tensor_to_boundary(
+        extrinsic_curvature, mesh, boundary_direction);
+    const auto& face_trace_extrinsic_curvature = dg::project_tensor_to_boundary(
+        trace_extrinsic_curvature, mesh, boundary_direction);
+    const auto& face_inv_jacobian =
+        dg::project_tensor_to_boundary(inv_jacobian, mesh, boundary_direction);
```

Comment:
Please add a note here that this projection could be avoided by using the `domain::Tags::DetSurfaceJacobian` that's in the DataBox, which is computed directly on the face (not projected). That would be better on Gauss meshes that have no grid point at the boundary.
The `DetSurfaceJacobian` can then be multiplied by the (conformal) metric determinant to form the area element. Note that the `DetSurfaceJacobian` is computed using the _conformal_ metric.